### ISSUE TITLE
[Bug] Android build fails when the build path contains regular expression meta characters

### ISSUE THREAD

**Issue by canatella** [firebase/firebase-cpp-sdk#523]
### [REQUIRED] Please fill in the following fields:

  * Pre-built SDK from the [website](https://firebase.google.com/download/cpp) or open-source from this repo: open-source
  * Firebase C++ SDK version: 8.1.0
  * Problematic Firebase Component: Auth 
  * Other Firebase Components in use: Auth, Firestore 
  * Platform you are using the C++ SDK on: Linux 
  * Platform you are targeting: Android

### [REQUIRED] Please describe the issue here:

When building for Android in a build path containing regular expression meta characters (in my case +), the build fails due to the expression at https://github.com/firebase/firebase-cpp-sdk/blob/8ad99ec49ff5d01e909d5cedc5366cf5325460ec/CMakeLists.txt#L199 interpreting them. 

#### Steps to reproduce:

Try to build the SDK for Android in `$HOME/+foobar/firebase-cpp-sdk` path for example.

**Comment by canatella** [comment]
Btw, I fixed it by using
```
  string(REPLACE "${CMAKE_CURRENT_LIST_DIR}" "" TMP ${FIREBASE_BINARY_DIR})
  string(REGEX REPLACE
     "/[^/]+/(.*)"
     "${CMAKE_CURRENT_LIST_DIR}/app/\\1"
     APP_BINARY_DIR "${TMP}")
```

**Comment by chkuang-g** [comment]
@canatella Thank you for reporting this!

I was able to reproduce the issue you mentioned by
1. `git clone` the repo to a folder with a `+`, ex. `+foo/`
2. Run `./gradlew :analytics:assembleRelease`

I'll let the team know and patch this.

**Comment by chkuang-g** [comment]
@canatella This should be fixed by #562.  I will close this ticket for now.  If you still experience the same error, please open a new one and reference this one (for visibility sake)

Shawn
